cmake/simdjson-flags.cmake

```diff
@@ -202,6 +202,13 @@ if(SIMDJSON_ENABLE_THREADS)
   set(THREADS_PREFER_PTHREAD_FLAG TRUE)
   find_package(Threads REQUIRED)
   target_link_libraries(simdjson-flags INTERFACE Threads::Threads)
+  ###
+  # Reasonably enough, if you compile simdjson with thread support, people
+  # who linked with the installed version will also need thread support. It
+  # would be really nice if CMake could automate it, but apparently it cannot.
+  ###
+  message (STATUS "The simdjson is compiled with thread support. Users who install this library and then invoke it from CMake might need to call find_package(Threads REQUIRED) before asking to link with the simdjson library, e.g., a line such as target_link_libraries(myprogram PRIVATE simdjson::simdjson) may need a prior line like find_package(Threads REQUIRED).")
```

Comment:
That is not how CMake should work. You are responsible to call `find_dependency(Threads)` in your `-config.cmake`
The idea is:
- Have a `<package>-config.cmake` which pulls in all required external/imported dependencies via `find_dependency`
- Let CMake export Targets as`<package>-Targets(-<config>).cmake` (Or however you decide to name it)
- Include the exported (configuration independent) target file `<package>-Targets.cmake` in `<package>-config.cmake`